src/nvim/buffer_defs.h

```diff
@@ -930,6 +932,10 @@ struct window_S {
 
   synblock_T  *w_s;                 /* for :ownsyntax */
 
+  int w_hl_id;                      ///< 'winhighlight' id
+  int w_hl_id_inactive;             ///< 'winhighlight' id for inactive window
```

Comment:
To support all builtin hl groups in the future, I suppose this will be some sort of list?